src/modules/chat_state/index.js

```diff
@@ -37,7 +40,16 @@ function loadHTML() {
         return;
     }
     if ($stateContainer.length) return;
-    $chatSettingsButtonSelector.before(CHAT_STATE_TEMPLATE);
+
+    if (
+        ($chatSettingsButtonSelector.parent().siblings('.tw-tooltip').length === 1)
+        && ($chatSettingsButtonSelector.parent().parent('.tw-tooltip-wrapper').length === 1)
+    ) {
+        $chatSettingsButtonSelector.parent().parent().before(CHAT_STATE_TEMPLATE);
+    } else {
+        // fallback to inserting directly before chat settings button if Twitch UI changes
+        $chatSettingsButtonSelector.before(CHAT_STATE_TEMPLATE);
+    }
```

Comment:
I added a check for the assumed tooltip DOM structure and a fallback so if the UI does change then it just goes back to how it's being inserted now (directly before the chat settings button).

> can you elaborate on this? iirc they don't really communicate them well at all other than the default chat text for slow mode.

emote only:
![](https://user-images.githubusercontent.com/7574985/67732911-18fd5680-f9ba-11e9-8e24-42eed932415f.png)
sub only:
![](https://user-images.githubusercontent.com/7574985/67733002-477b3180-f9ba-11e9-85ea-de7a79e8ca38.png)
slow mode (time only shown while slowed):
![](https://user-images.githubusercontent.com/7574985/67733333-4eef0a80-f9bb-11e9-9870-6642d07a243e.png)
r9k (only shows when trying to send a non-unique message):
![](https://user-images.githubusercontent.com/7574985/67733332-4e567400-f9bb-11e9-99d2-d45aa5777cd0.png)

The only information that isn't communicated by default is the slow mode duration before having sent a message and r9k mode if you haven't sent a non-unique message.

> i try hard to avoid adding extra toggles in settings, since it adds to the visual noise of too many choices a user can make.

Personally, when I added BTTV I didn't want much of the features at all and was only looking for the functionality of persisting deleted chat messages. When I first looked at the settings and saw all the toggles, I was impressed with the extent of the modularity and how it seemed that pretty much every feature could be turned off if I wanted to. The chat state icons, however, was the only feature that stood out to me as something that I couldn't disable.

IMO, the elegance in the consistency of the modularity of features and ability to turn on/off every feature in the extension outweighs the cost of visual noise of additional settings. (But of course that's a design decision that's up to you 🙂)